### PR TITLE
[RFC] vim-patch:8.0.0358,8.0.0359

### PR DESCRIPTION
**Pull request by ckelsel**
#### vim-patch:8.0.0358: invalid memory access in C-indent code

Problem:    Invalid memory access in C-indent code.
Solution:   Don't go over end of empty line. (Dominique Pelle, closes vim/vim#1492)

https://github.com/vim/vim/commit/60629d642541a089c322e65963c0a77e5f77eb79

#### vim-patch:8.0.0359: 'number' and 'relativenumber' are not properly tested

Problem:    'number' and 'relativenumber' are not properly tested.
Solution:   Add tests, change old style to new style tests. (Ozaki Kiichi,
            closes vim/vim#1447)

https://github.com/vim/vim/commit/dc9a081712ec8c140e6d4909e9f6b03a629d32d3

### CODE REVIEW

**Comment by ckelsel** [comment]
image

**Comment by ckelsel** [comment]
image

File: src/nvim/testdir/Makefile [hunk 1]
```diff
@@ -77,6 +77,7 @@ NEW_TESTS ?= \
 	    test_mksession_utf8.res \
 	    test_nested_function.res \
 	    test_normal.res \
+	    test_number.res \
```
Comment (justinmk) [inline src/nvim/testdir/Makefile]: should also add `test_findfile.res` in this Makefile, else (I think) running it as a single test won't work (`TEST_FILE=.... make oldtest`).

Comment (ckelsel) [reply src/nvim/testdir/Makefile]: @justinmk done.

File: test/functional/legacy/089_number_relnumber_findfile_spec.lua [hunk 2]
```diff
@@ -1,116 +0,0 @@
--- - Some tests for setting 'number' and 'relativenumber'
```
Comment (justinmk) [inline test/functional/legacy/089_number_relnumber_findfile_spec.lua]: Normally I recommend keeping the Lua-converted version of the legacy test, but this one is simple enough that it probably isn't useful.

Comment (ckelsel) [reply test/functional/legacy/089_number_relnumber_findfile_spec.lua]: @justinmk Ok, I will keep lua-converted version next time.

**Pull request merged**